### DISCUSSION TITLE
storage.from(bucket).remove([path]) doesn't work

### DISCUSSION THREAD

**Discussion by ashalfarhan** [supabase/supabase#2466] [Questions] [answered]
I don't know if this is a bug or something, but it seems like I can't remove a file on a bucket.
This only happens in my react-app (client), but it works if I execute this operation in node.js.

Here are the screenshots:
1. Client: 
- The code:
![image](https://user-images.githubusercontent.com/70800415/126382293-7e6929f4-2119-4219-add0-6861fb4c902a.png)
- Response: 
![image](https://user-images.githubusercontent.com/70800415/126382548-4f387915-3e32-40a3-b16a-1e987453c7cb.png)

2. Node.js:
- The code:
![image](https://user-images.githubusercontent.com/70800415/126382346-69476616-3594-4d52-af84-78c386817084.png)
- Response:
![image](https://user-images.githubusercontent.com/70800415/126382457-e0db949b-40ec-4763-ba30-ede1b77ba08a.png)

3. Supabase dashboard settings for bucket:
![image](https://user-images.githubusercontent.com/70800415/126383081-cf175aaf-e184-4144-9608-dce7d6f2b18a.png)

![image](https://user-images.githubusercontent.com/70800415/126383108-1bf4bf77-f15f-4e73-928a-7aa9b07688ef.png)

Note: Previously I've tried in `@supabase/supabase-js` ver.1.19.0, but after I upgrade to 1.20.0, this issue still face me

**Comment by inian** [comment]
Are you using the anonKey in the react app and serviceKey in the Node.js script @ashalfarhan? If so, this looks like issue with the RLS policies

**Reply by ashalfarhan** [reply]
No, I'm not, both are using the service key. Is there something wrong with my bucket configuration?

**Comment by zakaria-chahboun** [comment, marked as answer]
@ashalfarhan 
You must add select(`true`) and delete(`role() = 'anon':: text`) permissions under storage.objects.

### The docs said:
Policy permissions required:
* buckets permissions: none
* objects permissions: `delete` and `select`

Like this:
![Screenshot from 2021-09-10 14-39-56](https://user-images.githubusercontent.com/18173275/132862560-58968052-7798-4129-98f8-36d5dc444fa7.png)
